code/edgemedia/src/main/java/com/adobe/marketing/mobile/edge/media/internal/MediaEventProcessor.kt

```diff
@@ -11,18 +11,117 @@
 
 package com.adobe.marketing.mobile.edge.media.internal
 
-internal class MediaEventProcessor {
-    private val SOURCE_TAG = "MediaEventProcessor"
+import androidx.annotation.VisibleForTesting
+import com.adobe.marketing.mobile.Event
+import com.adobe.marketing.mobile.edge.media.internal.MediaInternalConstants.LOG_TAG
+import com.adobe.marketing.mobile.edge.media.internal.xdm.XDMMediaEvent
+import com.adobe.marketing.mobile.services.Log
+import java.util.UUID
 
+internal class MediaEventProcessor(
+    private val mediaState: MediaState,
+    private val dispatcher: (event: Event) -> Unit
+) {
+    private val sourceTag = "MediaEventProcessor"
+
+    @VisibleForTesting
+    internal val mediaSessions: MutableMap<String, MediaSession> = mutableMapOf()
+
+    /**
+     * Creates new [MediaSession], assigning it a new random ID.
+     * @return the session ID of the new [MediaSession]
+     */
+    fun createSession(): String {
+        val sessionId = UUID.randomUUID().toString()
```

Comment:
We should wrap all the our function bodies in mutex (similar to what we have in the [media](https://github.com/adobe/aepsdk-media-android/blob/main/code/media/src/main/java/com/adobe/marketing/mobile/media/internal/MediaRealTimeService.java#L105) project) to avoid any race conditions.